src/include/sof/audio/dcblock/dcblock.h

```diff
@@ -0,0 +1,84 @@
+#ifndef __SOF_AUDIO_DCBLOCK_DCBLOCK_H__
+#define __SOF_AUDIO_DCBLOCK_DCBLOCK_H__
+
+#include <stdint.h>
+#include <sof/platform.h>
+#include <ipc/stream.h>
+
+struct audio_stream;
+struct comp_dev;
+
+struct dcblock_state {
+	int32_t R_coeff; /**< Q1.31 */
+	int32_t x_prev; /**< state variable referring to x[n-1] */
+	int32_t y_prev; /**< state variable referring to y[n-1] */
+};
+
+struct sof_dcblock_config {
+/**
+ * array of R coefficients in Q1.31 for the filter. Each i-th entry
+ * corresponds to the value used for the i-th channel
+ */
+	int32_t R_coeffs[PLATFORM_MAX_CHANNELS];
```

Comment:
you have config in state and config is there a reason for that?

Reply:
From my understanding that's the same behavior as eq_iir and eq_fir. Although in those cases the config is a lot more complicated. 

Doing it this way, we ensure that for one call to copy(), all the audio processing is done with the same R coefficients.

Reply:
Rearranged the structure of the coefficients. Can you please take a look at it again?